vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/view/components/ProjectDetailsComponent.java

```diff
@@ -310,34 +311,112 @@ private void setupExperimentalDesignSearch() {
+            }
+            projectInformationService.manageProject(selectedProject.value(), value);
+          });
 
       ProjectDetailsComponent.Handler.submitOnValueChange(principalInvestigatorToggleComponent,
-          value -> projectInformationService.investigateProject(selectedProject.value(), value));
+          value ->
+          {
+            if (Objects.isNull(selectedProject)) {
+              return;
+            }
+            projectInformationService.investigateProject(selectedProject.value(), value);
+          });
       ProjectDetailsComponent.Handler.submitOnValueChange(responsiblePersonToggleComponent,
-          value -> projectInformationService.setResponsibility(selectedProject.value(), value));
-      //ToDo Store selection changes for species, specimen and analyte here
+          value ->
+          {
+            if (Objects.isNull(selectedProject)) {
+              return;
+            }
+            projectInformationService.setResponsibility(selectedProject.value(), value);
+          });
+      ProjectDetailsComponent.Handler.submitOnValueAdded(speciesMultiSelectComboBox,
+          value ->
+          {
+            if (Objects.isNull(selectedProject)) {
+              return;
+            }
+            projectInformationService.addSpeciesToActiveExperiment(selectedProject.value(),
+                value.toArray(Species[]::new));
+          });
+      ProjectDetailsComponent.Handler.submitOnValueAdded(specimenMultiSelectComboBox,
+          value ->
+          {
+            if (Objects.isNull(selectedProject)) {
+              return;
+            }
+            projectInformationService.addSpecimenToActiveExperiment(selectedProject.value(),
+                value.toArray(Specimen[]::new));
+          });
+
+      ProjectDetailsComponent.Handler.submitOnValueAdded(analyteMultiSelectComboBox,
+          value ->
+          {
+            if (Objects.isNull(selectedProject)) {
+              return;
+            }
+            projectInformationService.addAnalyteToActiveExperiment(selectedProject.value(),
+                value.toArray(Analyte[]::new));
+          });
     }
 
     private static <V, T extends HasValue<?, V>> void submitOnValueChange(T element,
         Consumer<V> submitAction) {
       element.addValueChangeListener(it -> submitAction.accept(element.getValue()));
     }
 
+    private static <V extends Collection<?>, T extends HasValue<?, V>> void submitOnValueAdded(
+        T element,
+        Consumer<V> submitAction) {
+      element.addValueChangeListener(it -> {
+        V oldValue = it.getOldValue();
+        V value = it.getValue();
+        if (oldValue.containsAll(value)) {
+          // nothing was added -> so we do not need to do anything
+        } else if (value.containsAll(oldValue)) {
+          // only added something
+          submitAction.accept(value);
+        } else {
+          //FIXME what to do? there seem to be elements deleted and added in this event
```

Comment:
What's the solution here?

Reply:
not yet determined. does not do anything for now. This needs a decision on how changes in the multi select combobox affect the stored state.